src/main/groovy/com/bmuschko/gradle/docker/DockerRemoteApiPlugin.groovy

```diff
@@ -46,28 +46,22 @@ class DockerRemoteApiPlugin implements Plugin<Project> {
     }
 
     private void configureAbstractDockerTask(Project project, DockerExtension extension) {
-        ThreadContextClassLoader dockerClassLoader = new DockerThreadContextClassLoader()
-
+        ThreadContextClassLoader dockerClassLoader = new DockerThreadContextClassLoader(extension, configurePluginClassPath(project))
         project.tasks.withType(AbstractDockerRemoteApiTask) {
-            Configuration config = project.configurations[DOCKER_JAVA_CONFIGURATION_NAME]
-
-            config.defaultDependencies { dependencies ->
-                dependencies.add(project.dependencies.create("com.github.docker-java:docker-java:$DockerRemoteApiPlugin.DOCKER_JAVA_DEFAULT_VERSION"))
-                dependencies.add(project.dependencies.create('org.slf4j:slf4j-simple:1.7.5'))
-                dependencies.add(project.dependencies.create('cglib:cglib:3.2.0'))
-            }
-
             group = DEFAULT_TASK_GROUP
             threadContextClassLoader = dockerClassLoader
-
-            conventionMapping.with {
-                classpath = { config }
-                url = { extension.url }
-                certPath = { extension.certPath }
-            }
         }
     }
 
+    private Set<File> configurePluginClassPath(Project project) {
+        project.repositories.addAll(project.buildscript.repositories.collect())
+        project.configurations.getByName(DOCKER_JAVA_CONFIGURATION_NAME).defaultDependencies { dependencies ->
+            dependencies.add(project.dependencies.create("com.github.docker-java:docker-java:$DockerRemoteApiPlugin.DOCKER_JAVA_DEFAULT_VERSION"))
+            dependencies.add(project.dependencies.create('org.slf4j:slf4j-simple:1.7.5'))
+            dependencies.add(project.dependencies.create('cglib:cglib:3.2.0'))
+        }.files
```

Comment:
The method call `.files` will resolve all dependencies right away during Gradle's configuration time. This should be avoided. Dependencies should only be resolved when they are actually used e.g. running a task.